### DISCUSSION TITLE
Redux

### DISCUSSION THREAD

**Discussion by daveOnactive** [vercel/next.js#17025] [Help] [answered]
I am trying to create a global state and I want to dispatch the action in the getStaticProps function, Please help

**Comment by leerob** [comment, marked as answer]
Here's a Redux example you can start with: https://github.com/vercel/next.js/tree/canary/examples/with-redux-toolkit

You probably don't want to (and can't) dispatch an action from `getStaticProps`, since it's run on the server (Node.js context). Instead, you're probably wanting to dispatch an action when the component first renders, yeah?

**Reply by daveOnactive** [reply]
Thanks

**Reply by ujjwal11** [reply]
why can't we dispatch an action in getStaticProps, getInitialProps also works on server side and we can dispatch an action on it.
